### PR TITLE
NAS-117953 / 22.12 / NAS-117953: Add 'serializeConfig' method

### PR DESCRIPTION
**Pull request by bvasilenko**
Fix for a bug when user clicks **Edit** and array fields are empty.

For testing, please see example in the ticket

### CODE REVIEW

File: src/app/services/app-schema.service.ts [hunk 1]
```diff
@@ -615,4 +615,27 @@ export class AppSchemaService {
...
+          valueConfig.forEach((valueItem, idxItem) => {
+            if (!_.isPlainObject(valueItem)) {
+              const keyItem = Object.keys(formConfig.value[idxItem])[0];
+              valueConfig[idxItem] = { [keyItem]: valueItem };
+            } else {
+              this.serializeConfig(valueItem, formConfig.controls[idxItem] as FormGroup);
+            }
+          });
+        } else if (_.isPlainObject(valueConfig)) {
+          this.serializeConfig(valueConfig, formConfig);
```
Comment (RehanY147) [inline src/app/services/app-schema.service.ts:636]: Not sure what this method is meant to do. Code needs to be more readable.

**Review by RehanY147 — CHANGES_REQUESTED** [review]

**Comment by bvasilenko** [comment]
@RehanY147 

> Shouldn't change the array while we're traversing it

Good one. Now it returns a modified value, and there's no mutable reference arguments

> Not sure what this method is meant to do. Code needs to be more readable.

Now I've renamed the method and added javadoc-style commentary

File: src/app/services/app-schema.service.ts [hunk 2]
```diff
@@ -615,4 +615,45 @@ export class AppSchemaService {
       return this.serializeFormValue(listItem[Object.keys(listItem)[0]]);
     }) as HierarchicalObjectMap<ChartFormValue>[];
   }
+
+  /**
+   * Restores keys from form group
+   * @param config Object without keys. Example: { objectList: [{ nestedList: ['test4', 'test5'] }] }
+   * @param form Form group to restore keys from
```
Comment (RehanY147) [inline src/app/services/app-schema.service.ts:622]: form Form group

File: src/app/services/app-schema.service.ts [hunk 3]
```diff
@@ -615,4 +615,45 @@ export class AppSchemaService {
...
+   * @returns Object with keys. Example: { objectList: [{ nestedList: ['test4', 'test5'] }] }
```
Comment (RehanY147) [inline src/app/services/app-schema.service.ts:623]: The example is the same as `@param config`

**Review by RehanY147 — APPROVED** [review]
Looks good. Nice!

**Pull request merged**